tests/foreman/ui/test_host.py

```diff
@@ -2331,18 +2360,32 @@ def test_positive_create_with_puppet_class(
         organization=module_puppet_org, location=module_puppet_loc
     )
     host_template.create_missing()
-
+    os_name = f'{host_template.operatingsystem.name} {host_template.operatingsystem.major}'
     with session_puppet_enabled_sat.ui_session as session:
         session.organization.select(org_name=module_puppet_org.name)
         session.location.select(loc_name='Any Location')
-        host_name = create_fake_host(
-            session,
-            host_template,
-            extra_values={
-                'host.puppet_environment': module_env_search.name,
-                'puppet_enc.classes.assigned': [module_import_puppet_module['puppet_class']],
-            },
+        host = make_fake_host(
```

Comment:
This needs to continue using the UI for the host creation - its a primary aspect of the test coverage here.